File: typescript-sensitiveinfodisclosure/secure8.ts
import OpenAI from "openai";

const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

async function getEnvironmentAdvice(): Promise<string> {
  const environment: string = process.env.NODE_ENV!;
  const response = await client.chat.completions.create({
    model: "gpt-4o",
    max_tokens: 512,
    messages: [
      { role: "user", content: `The app is running in ${environment} mode. What logging configuration is recommended?` }
    ],
  });
  return response.choices[0].message.content ?? "";
}

File: python-prompti/secure8.py
from flask import request
from openai import OpenAI

client = OpenAI()


def score_submitted_eval():
    sample = request.json["sample"]
    response = client.chat.completions.create(
        model="gpt-4o-mini",
        messages=[
            {"role": "system", "content": "Score this isolated test sample."},
            {"role": "user", "content": sample},
        ],
    )
    write_eval_artifact(response.choices[0].message.content)


def write_eval_artifact(_verdict):
    pass
